packages/core/src/selectors/base.ts

```diff
@@ -0,0 +1,74 @@
+import { QueryBundle, ToolMetadata } from "../types";
+
+export class SingleSelection {
```

Comment:
class is not initiated somewhere - use interface or type?

Reply:
agreed, updated